src/book/summary.rs

```diff
@@ -373,15 +421,31 @@ impl<'a> SummaryParser<'a> {
                 }
                 Some(Event::Start(Tag::List(..))) => {
                     // recurse to parse the nested list
-                    let (_, last_item) = get_last_link(&mut items)?;
-                    let last_item_number = last_item
-                        .number
-                        .as_ref()
-                        .expect("All numbered chapters have numbers");
-
-                    let sub_items = self.parse_nested_numbered(last_item_number)?;
-
-                    last_item.nested_items = sub_items;
+                    let (_, last_item) = get_last_linklike(&mut items)?;
+
+                    match *last_item {
+                        SummaryItem::Link(ref mut last_link) => {
+                            let last_item_number = last_link
+                                .number
+                                .as_ref()
+                                .expect("All numbered chapters have numbers");
+
+                            let sub_items = self.parse_nested_numbered(last_item_number)?;
+
+                            last_link.nested_items = sub_items;
+                        },
+                        SummaryItem::VirtualLink(ref mut last_link) => {
+                            let last_item_number = last_link
+                                .number
+                                .as_ref()
+                                .expect("All numbered chapters have numbers");
+
+                            let sub_items = self.parse_nested_numbered(last_item_number)?;
+
+                            last_link.nested_items = sub_items;
+                        },
+                        SummaryItem::Separator => unreachable!(),
+                    };
```

Comment:
I don't really like this match because of the code duplication. Some ideas on how to make it nicer?